src/core/ensure.rs

```diff
@@ -0,0 +1,15 @@
+use std::path::Path;
+use std::fs;
+use std::io;
+
+
+/// Ensure a directory exists and is empty
+pub fn ensure_dir_exists_fresh(dir: &str) -> io::Result<()> {
```

Comment:
Maybe expose a function that takes a Path directly so you don't need to convert back and forth as well.
Good place to put this helper though!